test/asynchronous/test_database.py

```diff
@@ -0,0 +1,778 @@
+            filter={"name": {"$regex": r"^(?!system\.)"}}
+        )
+        for coll in no_system_collections:
+            self.assertTrue(not coll.startswith("system."))
+        self.assertIn("systemcoll.test", no_system_collections)
+
+        # Force more than one batch.
+        db = self.client.many_collections
+        for i in range(101):
+            await db["coll" + str(i)].insert_one({})
+        # No Error
+        try:
+            await db.list_collection_names()
+        finally:
+            await self.client.drop_database("many_collections")
+
+    async def test_list_collection_names_filter(self):
+        listener = OvertCommandListener()
+        client = await async_rs_or_single_client(event_listeners=[listener])
+        db = client[self.db.name]
+        await db.capped.drop()
+        await db.create_collection("capped", capped=True, size=4096)
+        await db.capped.insert_one({})
+        await db.non_capped.insert_one({})
+        self.addAsyncCleanup(client.drop_database, db.name)
+        filter: Union[None, Mapping[str, Any]]
+        # Should not send nameOnly.
+        for filter in ({"options.capped": True}, {"options.capped": True, "name": "capped"}):
+            listener.reset()
+            names = await db.list_collection_names(filter=filter)
+            self.assertEqual(names, ["capped"])
+            self.assertNotIn("nameOnly", listener.started_events[0].command)
+
+        # Should send nameOnly (except on 2.6).
+        for filter in (None, {}, {"name": {"$in": ["capped", "non_capped"]}}):
+            listener.reset()
+            names = await db.list_collection_names(filter=filter)
+            self.assertIn("capped", names)
+            self.assertIn("non_capped", names)
+            command = listener.started_events[0].command
+            self.assertIn("nameOnly", command)
+            self.assertTrue(command["nameOnly"])
+
+    async def test_check_exists(self):
+        listener = OvertCommandListener()
+        client = await async_rs_or_single_client(event_listeners=[listener])
+        self.addAsyncCleanup(client.close)
+        db = client[self.db.name]
+        await db.drop_collection("unique")
+        await db.create_collection("unique", check_exists=True)
+        self.assertIn("listCollections", listener.started_command_names())
+        listener.reset()
+        await db.drop_collection("unique")
+        await db.create_collection("unique", check_exists=False)
+        self.assertTrue(len(listener.started_events) > 0)
+        self.assertNotIn("listCollections", listener.started_command_names())
+
+    async def test_list_collections(self):
+        await self.client.drop_database("pymongo_test")
+        db = AsyncDatabase(self.client, "pymongo_test")
+        await db.test.insert_one({"dummy": "object"})
+        await db.test.mike.insert_one({"dummy": "object"})
+
+        results = await (await db.list_collections()).to_list()
```

Comment:
`AsyncCursors` are asynchronous iterators, so you can use `[async for result in results]` instead of calling `to_list()` here.